src/mame/machine/k573dio.cpp

```diff
@@ -120,8 +133,15 @@ void k573dio_device::device_start()
 {
 	output_cb.resolve_safe();
 
-	ram = std::make_unique<uint16_t[]>(0x2000000/2);
-	save_pointer(NAME(ram), 0x2000000/2 );
+	const int ram_size = 0x2000000/2;
+	ram = std::make_unique<uint16_t[]>(ram_size);
```

Comment:
Could you please change this to a `memory_share_creator` in the DIO device, and then use either another `memory_share_creator` or a `required_memory_share` or another `memory_share_creator` in the FPGA device to find it?